File: sdk/node/src/deployments.ts
export class DeploymentsAPI {
  constructor(private client: any) {}

  async create(agentId: string, deploymentDef: any): Promise<any> {
    return this.client.request('POST', `/${agentId}/deployments`, deploymentDef);
  }

  async list(): Promise<any[]> {
    return this.client.request('GET', '/deployments');
  }

  async get(deploymentId: string): Promise<any> {
    return this.client.request('GET', `/deployments/${deploymentId}`);
  }

  async pause(deploymentId: string): Promise<any> {
    return this.client.request('POST', `/deployments/${deploymentId}/pause`);
  }

  async resume(deploymentId: string): Promise<any> {
    return this.client.request('POST', `/deployments/${deploymentId}/resume`);
  }

  async archive(deploymentId: string): Promise<any> {
    return this.client.request('POST', `/deployments/${deploymentId}/archive`);
  }

  async getUsage(deploymentId: string): Promise<any> {
    return this.client.request('GET', `/deployments/${deploymentId}/usage`);
  }

  async getSla(deploymentId: string): Promise<any> {
    return this.client.request('GET', `/deployments/${deploymentId}/sla`);
  }
}

File: sdk/python/kleberai/deployments.py
from typing import Any, Dict, List


class DeploymentsAPI:
    def __init__(self, client):
        self.client = client

    def create(self, agent_id: str, deployment_def: Dict[str, Any]) -> Dict[str, Any]:
        return self.client._request("POST", f"/{agent_id}/deployments", json=deployment_def)

    def list(self) -> List[Dict[str, Any]]:
        return self.client._request("GET", "/deployments")

    def get(self, deployment_id: str) -> Dict[str, Any]:
        return self.client._request("GET", f"/deployments/{deployment_id}")

    def update(self, deployment_id: str, deployment_def: Dict[str, Any]) -> Dict[str, Any]:
        return self.client._request("PATCH", f"/deployments/{deployment_id}", json=deployment_def)

    def pause(self, deployment_id: str) -> Dict[str, Any]:
        return self.client._request("POST", f"/deployments/{deployment_id}/pause")

    def resume(self, deployment_id: str) -> Dict[str, Any]:
        return self.client._request("POST", f"/deployments/{deployment_id}/resume")

    def archive(self, deployment_id: str) -> Dict[str, Any]:
        return self.client._request("POST", f"/deployments/{deployment_id}/archive")

    def rollback(self, deployment_id: str) -> Dict[str, Any]:
        return self.client._request("POST", f"/deployments/{deployment_id}/rollback")

    def get_usage(self, deployment_id: str) -> Dict[str, Any]:
        return self.client._request("GET", f"/deployments/{deployment_id}/usage")

    def get_sla(self, deployment_id: str) -> Dict[str, Any]:
        return self.client._request("GET", f"/deployments/{deployment_id}/sla")
